app/services/quarterly_external_income_export.rb

```diff
@@ -0,0 +1,58 @@
+class QuarterlyExternalIncomeExport
+  HEADERS = [
+    "RODA identifier",
+    "Delivery partner identifier",
+    "Delivery partner organisation",
+    "Title",
+    "Level",
+    "Providing organisation",
+    "ODA",
+  ]
+
+  def initialize(organisation, source_fund_code)
+    @organisation = organisation
+    @source_fund_code = source_fund_code
+    activity_ids = Activity.where(organisation_id: @organisation.id, source_fund_code: @source_fund_code).pluck(:id)
+    @activity_external_income = ExternalIncome.includes(:activity, :organisation).where(activity_id: activity_ids).order(:activity_id, :financial_year, :financial_quarter)
+    @financial_quarters = @activity_external_income.map(&:own_financial_quarter).uniq
```

Comment:
I prefer to not do anything that calls the database in an initializer, so here `activity_external_income` and `financial_quarters` would be private methods.

Reply:
I don't mind doing the changes, but I'd like to understand and learn why. ;)

Reply:
It's more of a feeling than a hard and fast rule TBH - the way I see it is that when calling a `call` method or similar, it should do everything at once. If you call an initializer and have it make a database query, you'll have to keep that data in memory for longer, whereas if you call it only when you need it, it can be thrown away quicker.